### PR TITLE
narwhalified MeanImputer

### PR DESCRIPTION
**Pull request by limlam96**
Converted transformer to narwhals https://github.com/lvgig/tubular/issues/344 #344

### CODE REVIEW

File: tests/base_tests.py [hunk 1]
```diff
@@ -585,7 +585,7 @@ def test_fit_not_changing_data_weighted(
     @pytest.mark.parametrize(
         "bad_weight_value, expected_message",
         [
-            (np.nan, "weight column must be non-null"),
```
Comment (davidhopkinson26) [inline tests/base_tests.py:588]: As in https://github.com/lvgig/tubular/pull/345#discussion_r1880067836

**Review by davidhopkinson26 — CHANGES_REQUESTED** [review]
in general looks good, but with the caveat of the issue with np.nan in the fit_not_changing_data_weighted test as we have seen in Mode and Median imputers.

File: tests/base_tests.py (hunk 1 above)
Comment (limlam96) [reply tests/base_tests.py]: pulled in the fix here :) https://github.com/lvgig/tubular/pull/351/commits/aed1e2c438dec42be95a2c1401c0648222165a2d

**Review by davidhopkinson26 — DISMISSED** [review]
Thanks @limlam96!

**Comment by limlam96** [comment]
> Sorry, just realised need to update CHANGELOG

ah yeah thanks, updated here https://github.com/lvgig/tubular/pull/351/commits/59d74fb1cb12b3c2c7b8dd3874fc4f88bbe70c29

**Review by davidhopkinson26 — APPROVED** [review]

**Pull request merged**